portal/eproms_substudy_tailored_content/src/js/Base.js

```diff
@@ -202,24 +202,50 @@ export default {
         getDefaultDomains() {
             return Object.keys(this.domainMappings);
         },
-        initTriggerDomains() {
+        initTriggerDomains(params) {
             if (!this.isTriggersNeeded()) {
+                setTimeout(function() {
+                    this.setInitView();
+                }.bind(this), 150);
                 return;
             }
+            params = params || {};
+            params.retryAttempt = params.retryAttempt || 0;
+            params.maxTryAttempts = params.maxTryAttempts || 5;
             //get trigger domains IF a patient
             Promise.allSettled([
                 this.$http(`/api/user/${this.getUserID()}/triggers`)
             ]).then(response => {
+                let processedData = "";
                 try {
-                    this.setUserDomains(JSON.parse(response[0].value));
-                    this.processDefaultDomainContent();
+                    processedData = JSON.parse(response[0].value);
                 } catch(e) {
                     //log error to console
                     this.setErrorMessage(`Error parsing trigger data ${e}`);
                 }
+
+                if (processedData &&
+                    params.retryAttempt < params.maxTryAttempts &&
+                    //if the trigger data has not been processed, try again until maximum number of attempts has been reached
```

Comment:
I may have overlooked it, but you should only bother to poll IF the state is "inprocess".  Checking against `processedTriggerStates` would also wait if the user is `unstarted` or `due`.

Reply:
right, the check won't be initiated if the triggers have been processed.